projects/mysadna2/src/karma.conf.js

```diff
@@ -0,0 +1,31 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/1.0/config/configuration-file.html
+
+module.exports = function (config) {
+  config.set({
+    basePath: '',
+    frameworks: ['jasmine', '@angular-devkit/build-angular'],
+    plugins: [
+      require('karma-jasmine'),
+      require('karma-chrome-launcher'),
+      require('karma-jasmine-html-reporter'),
+      require('karma-coverage-istanbul-reporter'),
+      require('@angular-devkit/build-angular/plugins/karma')
+    ],
+    client: {
+      clearContext: false // leave Jasmine Spec Runner output visible in browser
+    },
+    coverageIstanbulReporter: {
+      dir: require('path').join(__dirname, '../coverage'),
+      reports: ['html', 'lcovonly', 'text-summary'],
+      fixWebpackSourcePaths: true
+    },
+    reporters: ['progress', 'kjhtml'],
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: true,
+    browsers: ['Chrome'],
+    singleRun: false
+  });
+};
```

Comment:
It's good practice to end each file with new empty line.
You can easily fix it in `.ts` files by tslint. If you don't know what is tslint, I'll tell you later.
Other files (e.g. html, scss, txt etc) require more steps to find `end without new line`.
You can use git.
What I do:
When work is done and I commited all changes
1) `git show >show.txt`
2) Open show.txt by text editor
3) Find (Ctrl+F) word "newline"
Fix each file manually, where you see: `\ No newline at end of file`
4) Commit you fix
```
git add -A
git commit -m "endline fix"
```

Reply:
Don't forget to delete `show.txt`

Reply:
How do I 'fix' the files?